src/librbd/internal.cc

```diff
@@ -945,6 +986,188 @@ int validate_mirroring_enabled(ImageCtx *ictx) {
     return r;
   }
 
+  int create_cg(librados::IoCtx& io_ctx, const char *cg_name)
+  {
+    ostringstream bid_ss;
+    uint32_t extra;
+    int remove_r;
+    string id, id_cg, header_oid;
+
+    CephContext *cct = (CephContext *)io_ctx.cct();
+
+    Rados rados(io_ctx);
+    uint64_t bid = rados.get_instance_id();
+
+    int r = validate_pool(io_ctx, cct); // TODO Not sure if it's necessary at all. Consider removing
+    if (r < 0) {
+      return r;
+    }
+
+    id_cg = util::id_cg_name(cg_name);
+
+    r = io_ctx.create(id_cg, true);
+    if (r < 0) {
+      lderr(cct) << "error creating consistency group id object: " << cpp_strerror(r)
+		 << dendl;
+      return r;
+    }
+
+    extra = rand() % 0xFFFFFFFF;
+    bid_ss << std::hex << bid << std::hex << extra;
+    id = bid_ss.str();
```

Comment:
I copied this line from create image function. Where can I look at the UUID generator you are talking about?